tests/test_external_mixin.py

```diff
@@ -377,3 +421,146 @@ def test_restore_state(self, strategy):
+                                                      PVForecastExternalStrategy()], indirect=True)
+    def test_set_energy_measurement(self, ext_strategy_fixture):
+        # test successful call of set_energy_measurement:
+        transaction_id = str(uuid.uuid4())
+        arguments = {"transaction_id": transaction_id,
+                     "energy_measurement": {now().format(DATE_TIME_FORMAT): 1}}
+        payload = {"data": json.dumps(arguments)}
+        assert ext_strategy_fixture.pending_requests == deque([])
+        ext_strategy_fixture.set_energy_measurement(payload)
+        assert len(ext_strategy_fixture.pending_requests) > 0
+        energy_measurement_response_channel = f"{ext_strategy_fixture.channel_prefix}/" \
+                                              "response/set_energy_measurement"
+        assert (ext_strategy_fixture.pending_requests ==
+                deque([IncomingRequest("set_energy_measurement", arguments,
+                                       energy_measurement_response_channel)]))
+
+        # test if set_energy_measurement raises error when provided with wrong payload
+        ext_strategy_fixture.redis.publish_json = Mock()
+        ext_strategy_fixture.pending_requests = deque([])
+        payload = {"data": json.dumps({"transaction_id": transaction_id})}
+        ext_strategy_fixture.set_energy_measurement(payload)
+        ext_strategy_fixture.redis.publish_json.assert_called_with(
+            energy_measurement_response_channel, {"command": "set_energy_measurement",
+                                                  "error": "Incorrect _set_energy_measurement"
+                                                           " request. "
+                                                  "Available parameters: (energy_measurement).",
+                                                  "transaction_id": transaction_id})
+        assert len(ext_strategy_fixture.pending_requests) == 0
+
+    @pytest.mark.parametrize("ext_strategy_fixture", [LoadForecastExternalStrategy(),
+                                                      PVForecastExternalStrategy()], indirect=True)
+    def test_set_energy_forecast_impl(self, ext_strategy_fixture):
+        # test successful call of set_energy_forecast_impl:
+        transaction_id = str(uuid.uuid4())
+        ext_strategy_fixture.redis.publish_json = Mock()
+        arguments = {"transaction_id": transaction_id,
+                     "energy_forecast": {now().format(DATE_TIME_FORMAT): 1}}
+        response_channel = "response_channel"
+        ext_strategy_fixture.set_energy_forecast_impl(arguments, response_channel)
+        ext_strategy_fixture.redis.publish_json.assert_called_once_with(
+            response_channel, {"command": "set_energy_forecast",
+                               "status": "ready",
+                               "transaction_id": arguments["transaction_id"]})
+
+        # test if error is sent when wrong time format is provided
```

Comment:
I still think it is better if these are split into separate tests, nit again though